### ISSUE TITLE
No such entity with cartId = 958695

### ISSUE THREAD

**Issue by SolsWebdesign** [magmodules/magento2-channable#146]
Upgraded Channable for customer and now getting errors on certain (complete) orders (not all), not sure why. 
Error is:

1 exception(s):
<pre>
Exception #0 (Magento\Framework\Exception\NoSuchEntityException): No such entity
with cartId = 958695
#1 Magento\Quote\Model\QuoteRepository->loadQuote() called at
[vendor/magento/module-quote/Model/QuoteRepository.php:136]
#2 Magento\Quote\Model\QuoteRepository->get() called at
[vendor/magento/framework/Interception/Interceptor.php:58]
#3 Magento\Quote\Model\QuoteRepository\Interceptor->___callParent() called at
[vendor/magento/framework/Interception/Interceptor.php:138]
#4
Magento\Quote\Model\QuoteRepository\Interceptor->Magento\Framework\Interception\{closure}()
called at [vendor/magento/framework/Interception/Interceptor.php:153]
#5 Magento\Quote\Model\QuoteRepository\Interceptor->___callPlugins() called at
[generated/code/Magento/Quote/Model/QuoteRepository/Interceptor.php:23]
#6 Magento\Quote\Model\QuoteRepository\Interceptor->get() called at
[vendor/magmodules/magento2-channable/Block/Adminhtml/Order/Totals.php:54]
#7 Magmodules\Channable\Block\Adminhtml\Order\Totals->initTotals() called at
[vendor/magento/module-sales/Block/Order/Totals.php:64]
#8 Magento\Sales\Block\Order\Totals->_beforeToHtml() called at
[vendor/magento/framework/View/Element/AbstractBlock.php:1110]
</pre>

Ends up here:
Magmodules\Channable\Block\Adminhtml\Order\Totals.php

```
public function initTotals()
    {
        $quoteId = $this->getOrder()->getQuoteId();
        $quote = $this->quote->get($quoteId);
```

Seems to happen on orders that are already complete and customer ran into this when she tried to credit one of these. The quoteId is available but the next step "$quote = $this->quote->get($quoteId);" fails. Still investigating why but in order for the customer to continue her work I have made the following quick fix:

For now the quick fix is 

```
$quoteId = $this->getOrder()->getQuoteId();
        if(isset($quoteId) && $quoteId > 0) {
            try {
                $quote = $this->quote->get($quoteId);
            } catch (\Exception $exception) {
                // later / custom logging
            }
            if (isset($quote)) { ...
```

As soon as I have more information I will add it.

**Comment by SolsWebdesign** [comment]
Please note that the extension does not have to be enabled for this error to occur.

**Comment by roby0404** [comment]
I second this, was just about to report this same issue. @SolsWebdesign you are correct indeed, functionalities should not really rely on historical data from the quote entity itself due to the cron you mentioned, but rather the data such as the transaction_fee they attempt to fetch in the Totals.php class is supposed to be transferred onto identical sales_order attribute upon order creation.

**Comment by Marvin-Magmodules** [comment]
Hi @SolsWebdesign, @roby0404, we wil release an update later today for this issue!

**Comment by Marvin-Magmodules** [comment]
HI @SolsWebdesign, @roby0404,

We are happy to share that we've just released the new [v1.7.0](https://github.com/magmodules/magento2-channable/releases/tag/1.7.0) version where we covered this issue. Thank you for the detailed issue report.

We are closing this issue now but please feel free to reopen the issue if this still occurs.
